docs/sources/mimir/operators-guide/architecture/bucket-index/index.md

```diff
@@ -77,3 +77,5 @@ This circuit breaker ensures queriers and rulers do not return any partial query
 ## How it's used by the store-gateway
 
 The [store-gateway]({{< relref "../components/store-gateway.md" >}}), at startup and periodically, fetches the bucket index for each tenant that belongs to its shard, and uses it as the source of truth for the blocks and deletion marks in the storage. This removes the need to periodically scan the bucket to discover blocks belonging to its shard.
+
+[^1]: Ingesters regularly add new blocks to the bucket as they offload data to long-term storage, and compactors subsequently compact these blocks and mark the original blocks for deletion. Actual deletion happens after a delay (`compactor.deletion-delay`). Trying to fetch a block that has already been deleted will lead to query failures. Therefore, in this context, an _almost up-to-date_ view is a view that's outdated by less than `compactor.deletion-delay`.
```

Comment:
```suggestion
[^1]: Ingesters regularly add new blocks to the bucket as they offload data to long-term storage, and compactors subsequently compact these blocks and mark the original blocks for deletion. Actual deletion happens after a delay (`-compactor.deletion-delay`). Trying to fetch a block that has already been deleted will lead to query failures. Therefore, in this context, an _almost up-to-date_ view is a view that's outdated by less than `-compactor.deletion-delay`.
```

Reply:
> in this context

is meant to refer to the use of the bucket index by `querier`s, `store-gateway`s and `ruler`s (that's what the sentence in the doc is referring to). Do these components vary in their use of the bucket index with respect to how much out of date the bucket index can be without causing failures?

Reply:
> I find it a bit difficult precisely describe what almost up-to-date means because it depends on the actual context.

If you find it difficult to describe precisely I think that's even more reason to pick the different meanings apart and describe them well for each context they're used in.